tasks/rulegen/src/main.rs

```diff
@@ -175,6 +175,39 @@ fn format_code_snippet(code: &str) -> String {
     format!("r#\"{}\"#", code.replace("\\\"", "\""))
 }
 
+fn dedent(s: &str) -> String {
+    let lines: Vec<&str> = s.lines().collect();
+    if lines.is_empty() {
+        return String::new();
+    }
+    // find the min whitespace count
+    let min_indent = lines
+        .iter()
+        .filter(|line| !line.trim().is_empty())
+        .map(|line| line.chars().take_while(|c| c.is_whitespace()).count())
+        .min()
+        .unwrap_or_default();
+
+    lines
+        .iter()
+        .map(|line| if line.len() >= min_indent { &line[min_indent..] } else { line })
+        .collect::<Vec<&str>>()
```

Comment:
type can be inferred:

```suggestion
        .collect::<Vec<_>>()
```